arquillian-junit5/src/main/java/org/wildfly/guide/testing/resources/TaskResource.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.guide.testing.resources;
+
+import java.time.Instant;
+import java.util.Collection;
+
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.Consumes;
+import jakarta.ws.rs.DELETE;
+import jakarta.ws.rs.GET;
+import jakarta.ws.rs.NotFoundException;
+import jakarta.ws.rs.PATCH;
+import jakarta.ws.rs.POST;
+import jakarta.ws.rs.PUT;
+import jakarta.ws.rs.Path;
+import jakarta.ws.rs.PathParam;
+import jakarta.ws.rs.Produces;
+import jakarta.ws.rs.core.MediaType;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.core.UriInfo;
+
+import org.wildfly.guide.testing.data.TaskRegistry;
+import org.wildfly.guide.testing.model.Task;
+
+/**
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@Path("task")
+@RequestScoped
+public class TaskResource {
+
+    @Inject
+    private TaskRegistry taskRegistry;
+    @Inject
+    private UriInfo uriInfo;
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Collection<Task> allTests() {
+        return taskRegistry.getTasks();
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/completed/{completed}")
+    public Collection<Task> getAllCompleted(@PathParam("completed") final boolean completed) {
+        return taskRegistry.getTasks(completed);
+    }
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/{id}")
+    public Task getTask(@PathParam("id") final long id) {
+        return taskRegistry.getTaskById(id);
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response addTask(final Task task) {
+        task.setAdded(Instant.now());
+        final Task added = taskRegistry.add(task);
+        return Response
+                .created(uriInfo.getBaseUriBuilder().path("task/" + added.getId()).build())
+                .build();
+    }
+
+    @PUT
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response editTask(final Task task) {
+        task.setUpdated(Instant.now());
+        final Task added = taskRegistry.updateOrAdd(task);
+        if (added == null) {
+            return Response.noContent().location(uriInfo.getBaseUriBuilder().path("task/" + task.getId()).build()).build();
```

Comment:
Maybe return 404 instead of no content?

Reply:
Actually, this is correct. When `null` is returned, the content was updated. This should happen per https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT#examples. 

> If the target resource does have a current representation and that representation is successfully modified with the state in the request, the origin server must send either a [200 OK](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200) or a [204 No Content](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204) to indicate successful completion of the request: